python/py-ttfquery/Portfile

```diff
@@ -19,20 +19,16 @@ long_description    TTFQuery builds on the FontTools package to allow the\
 
 homepage            http://ttfquery.sourceforge.net/
 
-master_sites        pypi:T/${real_name}/
+checksums           rmd160  fe4f7f20d218c59e41e9085851680978787e4686 \
+                    sha256  18e18eb03c18ea7dd08ed1b369adfd55c3815e34b69c9e321659b0c9f5576e44 \
+                    size    18434
 
-distname            ${real_name}-${version}
-
-checksums           rmd160 5630dbbda57708d4170bcba757c669c1a1b60e69 \
-                    sha256 d5b8d369903ee2754541819f27de8ea35486d124484e36a4869503d9a9ac7e4d
-
-python.versions     27
+python.versions     27 37 38 39
 
 if {$subport ne $name} {
-    depends_build-append    port:fonttools
+    depends_build-append \
+                    port:fonttools \
```

Comment:
with the recent changes to the `fonttools` port, upon your request, you should now likely depend on `port:py${python.version}-fonttools`

Reply:
upstream also lists `numpy` as a dependency in their `setup.py`: `install_requires = ['fonttools','numpy'],`

Also, I would assume that numpy and fonttools are not only needed at build time but also at runtime, correct? In that case they should be listed as `depends_lib`.

Reply:
the port also installs `console_scripts` and if you check, for example, `/opt/local/bin/ttffamily-3.9` you'll see that it uses `importlib.metadata`. That is part of Python since 3.8+ but for 2.7 and 3.8 you should add a dependency on the `py-importlib-metadata` port.

Reply:
@reneeotten thanks, tooo many ports and I've missed. I thought it was needed only at beet's subports.

About dependencies: I've moved to `depends_lib` as you suggested. Anyway,  the script works without `py-importlib-metadata`
```
√ macports-ports % sudo port uninstall py37-importlib-metadata                                                                      (git)-[-py-ttfquery]
√ macports-ports % /opt/local/bin/ttffamily-3.7                                                                                     (git)-[-py-ttfquery]
INFO:ttfquery.ttffiles:Registry already populated
INFO:ttfquery.ttffiles:166 fonts available
F github-octicons
  /Users/catap/Library/Fonts/octicons.ttf
F Weather Icons
  /Users/catap/Library/Fonts/weathericons.ttf
F all-the-icons
  /Users/catap/Library/Fonts/all-the-icons.ttf
F FontAwesome
  /Users/catap/Library/Fonts/fontawesome.ttf
F file-icons
  /Users/catap/Library/Fonts/file-icons.ttf
...
````
and when I've installed it:
```
√ macports-ports % sudo port install --unrequested py37-importlib-metadata                                                          (git)-[-py-ttfquery]
--->  Computing dependencies for py37-importlib-metadata
The following dependencies will be installed: 
 py37-typing_extensions
 py37-zipp
Continue? [Y/n]: y
--->  Fetching archive for py37-typing_extensions
--->  Attempting to fetch py37-typing_extensions-3.10.0.2_0.darwin_20.x86_64.tbz2 from https://cph.dk.packages.macports.org/py37-typing_extensions
--->  Attempting to fetch py37-typing_extensions-3.10.0.2_0.darwin_20.x86_64.tbz2.rmd160 from https://cph.dk.packages.macports.org/py37-typing_extensions
--->  Installing py37-typing_extensions @3.10.0.2_0
--->  Activating py37-typing_extensions @3.10.0.2_0
--->  Cleaning py37-typing_extensions
--->  Fetching archive for py37-zipp
--->  Attempting to fetch py37-zipp-3.5.0_0.darwin_20.x86_64.tbz2 from https://cph.dk.packages.macports.org/py37-zipp
--->  Attempting to fetch py37-zipp-3.5.0_0.darwin_20.x86_64.tbz2.rmd160 from https://cph.dk.packages.macports.org/py37-zipp
--->  Installing py37-zipp @3.5.0_0
--->  Activating py37-zipp @3.5.0_0
--->  Cleaning py37-zipp
--->  Fetching archive for py37-importlib-metadata
--->  Attempting to fetch py37-importlib-metadata-4.6.4_0.darwin_20.noarch.tbz2 from https://cph.dk.packages.macports.org/py37-importlib-metadata
--->  Attempting to fetch py37-importlib-metadata-4.6.4_0.darwin_20.noarch.tbz2.rmd160 from https://cph.dk.packages.macports.org/py37-importlib-metadata
--->  Installing py37-importlib-metadata @4.6.4_0
--->  Activating py37-importlib-metadata @4.6.4_0
--->  Cleaning py37-importlib-metadata
--->  Scanning binaries for linking errors
--->  No broken files found.                             
--->  No broken ports found.
√ macports-ports % /opt/local/bin/ttffamily-3.7                                                                                     (git)-[-py-ttfquery]
INFO:ttfquery.ttffiles:Registry already populated
INFO:ttfquery.ttffiles:166 fonts available
F github-octicons
  /Users/catap/Library/Fonts/octicons.ttf
F Weather Icons
  /Users/catap/Library/Fonts/weathericons.ttf
F all-the-icons
  /Users/catap/Library/Fonts/all-the-icons.ttf
F FontAwesome
  /Users/catap/Library/Fonts/fontawesome.ttf
F file-icons
  /Users/catap/Library/Fonts/file-icons.ttf
F Material Icons
  /Users/catap/Library/Fonts/material-design-icons.ttf
F Symbol
  /System/Library/Fonts/Symbol.ttf
F .SF NS Mono
  /System/Library/Fonts/SFNSMono.ttf
  /System/Library/Fonts/SFNSMonoItalic.ttf
F Apple Braille

```

Reply:
@reneeotten you're right. Thanks!